src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -593,11 +593,18 @@ protected override void ProcessRecord()
                             }
 
                             // Detect insecure redirection
-                            if (!AllowInsecureRedirect && response.RequestMessage.RequestUri.Scheme == "https" && response.Headers.Location?.Scheme == "http")
+                            if (!AllowInsecureRedirect)
                             {
-                                ErrorRecord er = new(new InvalidOperationException(), "InsecureRedirection", ErrorCategory.InvalidOperation, request);
-                                er.ErrorDetails = new ErrorDetails(WebCmdletStrings.InsecureRedirection);
-                                ThrowTerminatingError(er);
+                                // Skip insecure redirection detection if the URIs are relative 
```

Comment:
Can you please add comments about
1. why we are skipping the detection if the URIs are relative?
2. what is the expected behavior when either URI is relative?

Reply:
1) We are skipping detection if the URIs are relative, because `get_Scheme` is not supported on relative URIs
Error from #19461:
```
Type       : System.InvalidOperationException
    TargetSite : 
        Name          : get_Scheme
        DeclaringType : uri
        MemberType    : Method
        Module        : System.Private.Uri.dll
    Message    : This operation is not supported for a relative URI.
```
2) When IWR/IRM is given a relative uri it adds http (`WebRequestPSCmdlet.Common.cs` line 1406), so it's no longer a relative uri (for this reason I don't think the origin uri can be relative, but we should check anyway).
For this reason I think we should expect a silent failure on a https to no scheme uri redirect (https to http redirects are forbidden)
Unfortunately I don't know exactly what happens with the many proxies described in issue #19461, but it seems checking for the scheme creates a breaking error

Reply:
Thanks @CarloToso! Can you please incorporate the following 2 points into the comment in the code?
- We are skipping detection if the URIs are relative, because `Scheme` property is not supported on relative URIs
- If either URI is relative, then we skip the check and there may be a failure due to https-to-http redirects being forbidden by default.

Reply:
I also asked the issue author to verify the fix using the private build from the CIs of this PR.